lib/src/Base/Func/SymbolicParserExprTk.cxx

```diff
@@ -228,15 +228,19 @@ SymbolicParserExprTk::ExpressionCollection SymbolicParserExprTk::allocateExpress
   symbol_table.add_function("besselY1", y1);
   for (UnsignedInteger inputIndex = 0; inputIndex < inputDimension; ++ inputIndex)
   {
+    const String varName(inputVariablesNames_[inputIndex]);
+    if (symbol_table.symbol_exists(varName))
+      throw InvalidArgumentException(HERE) << "Variable name '" << varName << "' clobbers an existing symbol";
     if (!symbol_table.add_variable(inputVariablesNames_[inputIndex], stack[inputIndex]))
```

Comment:
@jschueller 

Minor nit: Could `varName `be used here instead of `inputVariablesNames_[inputIndex]`